lightning/src/chain/package.rs

```diff
@@ -380,10 +413,75 @@ impl PackageSolvingData {
 		}
 		true
 	}
-	fn get_finalized_tx<Signer: Sign>(&self, outpoint: &BitcoinOutPoint, onchain_handler: &mut OnchainTxHandler<Signer>) -> Option<Transaction> {
+	fn get_finalized_tx<Signer: Sign, U: Deref>(&self, outpoint: &BitcoinOutPoint, onchain_handler: &mut OnchainTxHandler<Signer>, value: u64, destination_script: Script, utxo_pool: &U) -> Option<Vec<Transaction>>
+		where U::Target: UtxoPool
+	{
 		match self {
-			PackageSolvingData::HolderHTLCOutput(ref outp) => { return onchain_handler.get_fully_signed_htlc_tx(outpoint, &outp.preimage); }
-			PackageSolvingData::HolderFundingOutput(ref outp) => { return Some(onchain_handler.get_fully_signed_holder_tx(&outp.funding_redeemscript)); }
+			PackageSolvingData::HolderHTLCOutput(ref outp) => {
+				if let Some(signed_htlc_tx) = onchain_handler.get_fully_signed_htlc_tx(outpoint, &outp.preimage) {
+					return Some(vec![signed_htlc_tx]);
+				}
+				return None;
+			},
+			PackageSolvingData::HolderFundingOutput(ref outp) => {
+				// We sign our commitment transaction
+				let mut package_txn = vec![];
+				let signed_tx = onchain_handler.get_fully_signed_holder_tx(&outp.funding_redeemscript);
+
+				if let Some(ref bumping_outpoint) = outp.utxo_input {
```

Comment:
Do we gain material flexibility by not building the transaction for the user and instead just giving them the output info (and relevant utilities needed to implement transaction construction)? That would simplify our code here significantly, I think?

If we don't gain any flexibility, then I think I'm happy with the general structure.